zebROS_ws/src/controllers_2023/src/elevator_controller.cpp

```diff
@@ -0,0 +1,443 @@
+    },
+    "Elevator Zeroing Timeout");
+    ddr_.registerVariable<double>
+    ("stage_2_height",
+     [this]()
+    {
+        return stage_2_height.load();
+    },
+    [this](double b)
+    {
+        stage_2_height.store(b);
+    },
+    "Stage 2 Height");
+    ddr_.registerVariable<double>
+    ("motion_magic_velocity_fast",
+     [this]()
+    {
+        return motion_magic_velocity_fast.load();
+    },
+    [this](double b)
+    {
+        motion_magic_velocity_fast.store(b);
+    },
+    "fast Motion Magic Velocity");
+    ddr_.registerVariable<double>
+    ("motion_magic_acceleration_fast",
+     [this]()
+    {
+        return motion_magic_acceleration_fast.load();
+    },
+    [this](double b)
+    {
+        motion_magic_acceleration_fast.store(b);
+    },
+    "Fast Motion Magic Acceleration");
+    ddr_.registerVariable<int>
+    ("motion_s_curve_strength",
+     [this]()
+    {
+        return motion_s_curve_strength.load();
+    },
+    [this](int b)
+    {
+        motion_s_curve_strength.store(b);
+    },
+    "S Curve Strength");
+    ddr_.publishServicesTopics();
+
+
+    if (!readIntoScalar(controller_nh, "arb_feed_forward_up_low", arb_feed_forward_up_low))
+    {
+        ROS_ERROR("Could not find arb_feed_forward_high");
+        return false;
+    }
+
+    if (!readIntoScalar(controller_nh, "arb_feed_forward_up_high", arb_feed_forward_up_high))
+    {
+        ROS_ERROR("Could not find arb_feed_forward_low");
+        return false;
+    }
+
+    if (!readIntoScalar(controller_nh, "elevator_zeroing_percent_output", elevator_zeroing_percent_output))
+    {
+        ROS_ERROR("Could not find elevator_zeroing_percent_output");
+        return false;
+    }
+
+    if (!readIntoScalar(controller_nh, "elevator_zeroing_timeout", elevator_zeroing_timeout))
+    {
+        ROS_ERROR("Could not find elevator_zeroing_timeout");
+        return false;
+    }
+
+    if (!readIntoScalar(controller_nh, "stage_2_height", stage_2_height))
+    {
+        ROS_ERROR("Could not find stage_2_height");
+        return false;
+    }
+
+    if (!readIntoScalar(controller_nh, "motion_magic_velocity", motion_magic_velocity_fast))
+    {
+        ROS_ERROR("Could not find motion_magic_velocity");
+        return false;
+    }
+
+
+    if (!readIntoScalar(controller_nh, "motion_magic_acceleration", motion_magic_acceleration_fast))
+    {
+        ROS_ERROR("Could not find motion_magic_acceleration");
+        return false;
+    }
+
+
+    if (!readIntoScalar(controller_nh, "motion_s_curve_strength", motion_s_curve_strength))
+    {
+    	ROS_ERROR("Could not find motion_s_curve_strength");
+    	return false;
+    }
+
+
+    //get config values for the elevator talon
+    XmlRpc::XmlRpcValue elevator_params;
+    if (!controller_nh.getParam("elevator_joint", elevator_params))
+    {
+        ROS_ERROR("Could not find elevator_joint");
+        return false;
+    }
+
+    if (!controller_nh.getParam("MAX_HEIGHT_VAL", MAX_HEIGHT_VAL))
```

Comment:
Nit picking : make both the param name string and the variable name lower case to match the rest of them